ddl/ddl_db_test.go

```diff
@@ -571,3 +572,48 @@ func (s *testDBSuite) TestUpdateMultipleTable(c *C) {
 
 	tk.MustQuery("select * from t1").Check(testkit.Rows("8 1 9", "8 2 9"))
 }
+
+func (s *testDBSuite) TestTruncateTable(c *C) {
+	defer testleak.AfterTest(c)
+	store, err := tidb.NewStore("memory://truncate_table")
+	c.Assert(err, IsNil)
+	tk := testkit.NewTestKit(c, store)
+	tk.MustExec("use test")
+	tk.MustExec("create table t (c1 int, c2 int)")
+	tk.MustExec("insert t values (1, 1), (2, 2)")
+	ctx := tk.Se.(context.Context)
+	is := sessionctx.GetDomain(ctx).InfoSchema()
+	oldTblInfo, err := is.TableByName(model.NewCIStr("test"), model.NewCIStr("t"))
+	c.Assert(err, IsNil)
+	oldTblID := oldTblInfo.Meta().ID
+
+	tk.MustExec("truncate table t")
+
+	tk.MustExec("insert t values (3, 3), (4, 4)")
+	tk.MustQuery("select * from t").Check(testkit.Rows("3 3", "4 4"))
+
+	is = sessionctx.GetDomain(ctx).InfoSchema()
+	newTblInfo, err := is.TableByName(model.NewCIStr("test"), model.NewCIStr("t"))
+	c.Assert(newTblInfo.Meta().ID, Greater, oldTblID)
+
+	// verify that the old table data has been deleted by background worker.
```

Comment:
Is there any chance that bg worker do not have enough time to delete the old table data?